pkg/charts/linkerd2/values_test.go

```diff
@@ -167,14 +179,24 @@ func TestNewValues(t *testing.T) {
 	}
 
 	t.Run("HA", func(t *testing.T) {
+		haUnavailable := intstr.FromString("1")
+		haDeploymentStrategy := &appv1.DeploymentStrategy{
+			RollingUpdate: &appv1.RollingUpdateDeployment{
+				MaxUnavailable: &haUnavailable,
+				MaxSurge:       &defaultSurge,
+			},
+		}
 		err := MergeHAValues(actual)
+		actual.DeploymentStrategy = haDeploymentStrategy
```

Comment:
This isn't ready yet. Trying to figure out why this is required in order for the tests to pass. The merge logic doesn't seem to match the helm template debug  output here but may be something I am missing. 👍